com.unity.render-pipelines.universal/Shaders/SimpleLit.shader

```diff
@@ -125,6 +125,12 @@ Shader "Universal Render Pipeline/Simple Lit"
             #pragma multi_compile_instancing
             #pragma multi_compile _ DOTS_INSTANCING_ON
 
+            // -------------------------------------
+            // Universal Pipeline keywords
+
+            // This is used during shadow map generation to differentiate between directional and punctual light shadows, as they use different formulas to apply Normal Bias
+            #pragma multi_compile_vertex _ _CASTING_PUNCTUAL_LIGHT_SHADOW
```

Comment:
do we need to add this pragma to shader graph generation?

Reply:
Yes, this pragma is needed in shadow-casting shader passes, in order to have Normal Bias applied correctly for punctual light shadows.

In revision 05ea77e I updated shader graph code that generates shadow caster pass shader, to make it include this keyword.